pennylane/_device.py

```diff
@@ -175,6 +174,18 @@ def observables(self):
         """
         raise NotImplementedError
 
+    def set_shots(self, shots):
+        """Changes the number of shots.
+
+        Args:
+            shots (int): number of circuit evaluations/random samples used to estimate
+            expectation values of observables.
+        """
+        if shots < 1:
+            raise DeviceError("The specified number of shots needs to be at least 1. Got {}.".format(shots))
+
+        self.shots = shots
```

Comment:
```suggestion
        self.shots = int(shots)
```
We could also do input validation that shots is integral, but lets instead take advantage of duck typing :laughing:

Reply:
I actually did `self.shots = shots` because that was what we used before in the `Device.__init__`, but duck typing is fine!

Reply:
Ah! No worries, sometimes code review can make you critically analyze decisions you forgot you made earlier :laughing: